torch/csrc/jit/passes/quantization/insert_observers.cpp

```diff
@@ -457,36 +459,60 @@ graph(%input, %weight, %bias, %4):
+graph(%self, %a, %b, %alpha, %relu):
+     %first_output = aten::add(%a, %b, %alpha)
+     %second_output = prim::CallMethod[name="forward.*"](%relu, %first_output)
+     return (%second_output) )",
+      {aten_add_alpha_is_one, is_relu_module});
+
+  const PatternInfo add_f_relu = PatternInfo::parse_from_str(
+      R"(
+graph(%self, %a, %b, %alpha, %relu, %inplace):
+     %first_output = aten::add(%a, %b, %alpha)
+     %second_output = prim::CallFunction(%relu, %first_output, %inplace)
+     return (%second_output) )",
+      {aten_add_alpha_is_one, is_functional_relu});
+
+  const PatternInfo inplace_add_nn_relu = PatternInfo::parse_from_str(
+      R"(
+graph(%self, %a, %b, %alpha, %relu):
+     %first_output = aten::add_(%a, %b, %alpha)
+     %second_output = prim::CallMethod[name="forward.*"](%relu, %first_output)
+     return (%second_output) )",
+      {aten_add_alpha_is_one, is_relu_module});
+
+  const PatternInfo inplace_add_f_relu = PatternInfo::parse_from_str(
+      R"(
+graph(%self, %a, %b, %alpha, %relu, %inplace):
+     %first_output = aten::add_(%a, %b, %alpha)
+     %second_output = prim::CallFunction(%relu, %first_output, %inplace)
+     return (%second_output) )",
+      {aten_add_alpha_is_one, is_functional_relu});
+
+  const PatternInfo add_aten_relu = PatternInfo::parse_from_str(R"(
+graph(%self, %a, %b, %alpha):
+     %first_output = aten::add(%a, %b, %alpha)
+     %second_output = aten::relu(%first_output)
      return (%second_output) )");
 
-  const PatternInfo add_f_relu = PatternInfo::parse_from_str(R"(
-graph(%self, %a, %b, %inplace):
-     %one = prim::Constant[value=1]()
-     %first_output = aten::add(%a, %b, %one)
-     %relu = prim::Constant[name="relu"]()
-     %second_output = prim::CallFunction(%relu, %first_output, %inplace)
+  const PatternInfo add_aten_relu_ = PatternInfo::parse_from_str(R"(
+graph(%self, %a, %b, %alpha):
+     %first_output = aten::add(%a, %b, %alpha)
+     %second_output = aten::relu_(%first_output)
      return (%second_output) )");
 
-  const PatternInfo inplace_add_nn_relu = PatternInfo::parse_from_str(R"(
-graph(%self, %a, %b):
-     %one = prim::Constant[value=1]()
-     %first_output = aten::add_(%a, %b, %one)
-     %second_module = match::module[name="ReLU"](%self)
-     %second_output = prim::CallMethod[name="forward"](%second_module, %first_output)
+  const PatternInfo inplace_add_aten_relu = PatternInfo::parse_from_str(R"(
+graph(%self, %a, %b, %alpha):
+     %first_output = aten::add_(%a, %b, %alpha)
+     %second_output = aten::relu(%first_output)
      return (%second_output) )");
 
-  const PatternInfo inplace_add_f_relu = PatternInfo::parse_from_str(R"(
-graph(%self, %a, %b, %inplace):
-     %one = prim::Constant[value=1]()
-     %first_output = aten::add_(%a, %b, %one)
-     %relu = prim::Constant[name="relu"]()
-     %second_output = prim::CallFunction(%relu, %first_output, %inplace)
+  const PatternInfo inplace_add_aten_relu_ = PatternInfo::parse_from_str(R"(
```

Comment:
Can we generate these patterns to make the code readable? If you iterate over the patterns (add, add_) and (relu, relu_) can you simplify this part?

Reply:
this is the third starter task for my intern, so let's land this asap